src/Data/Bifunctor/TH.hs

```diff
@@ -1143,6 +1148,8 @@ functorLikeTraverse tvMap FT { ft_triv = caseTrivial,     ft_var = caseVar
           -> tuple $ Unboxed len
           |  fc || or (take numFirstArgs xcs)
           -> wrongArg                    -- T (..var..)    ty_1 ... ty_n
+          | numLastArgs > 2
+          -> outOfPlaceTyVarError conName
```

Comment:
By definition, none of the `numFirstArgs` elements of `xcs` contain occurrences of the last two type parameters, so we no longer need to check `or (take numFirstArgs xcs)` here. I'd propose writing this like so:

```hs
          |  fc || numLastArgs > 2
          -> wrongArg                    -- T (..var..)    ty_1 ... ty_n
```

Moreover, you don't need to call `outOfPlaceTyVarError conName`, as `wrongArg` is meant to abstract over how each derived class chooses to deal with occurrences of out-of-place type variables. (This avoids having to pass `conName` to `functorLikeTraverse` in the first place.)